examples/connext_dds/asynchronous_publication_statistic/c/async_publisher.c

```diff
@@ -0,0 +1,268 @@
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize topic QoS, use
+       the configuration file USER_QOS_PROFILES.xml */
+    topic = DDS_DomainParticipant_create_topic(
+            participant,
+            "Example async",
+            type_name,
+            &DDS_TOPIC_QOS_DEFAULT,
+            NULL /* listener */,
+            DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        printf("create_topic error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize data writer QoS, use
+       the configuration file USER_QOS_PROFILES.xml */
+    writer = DDS_Publisher_create_datawriter(
+            publisher,
+            topic,
+            &DDS_DATAWRITER_QOS_DEFAULT,
+            NULL /* listener */,
+            DDS_STATUS_MASK_NONE);
+    if (writer == NULL) {
+        printf("create_datawriter error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+    async_writer = asyncDataWriter_narrow(writer);
+    if (async_writer == NULL) {
+        printf("DataWriter narrow error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Create data sample for writing */
+    instance = asyncTypeSupport_create_data_ex(DDS_BOOLEAN_TRUE);
+    if (instance == NULL) {
+        printf("asyncTypeSupport_create_data error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Create the data to be written, ensuring it is larger than message_size_max */
+    if (!DDS_OctetSeq_ensure_length(&instance->data, 8000, 8000)) {
+        printf("DDS_OctetSeq_ensure_length error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* For a data type that has a key, if the same instance is going to be
+       written multiple times, initialize the key here
+       and register the keyed instance prior to writing */
+    /*
+        instance_handle = asyncDataWriter_register_instance(
+            async_writer, instance);
+    */
+
+
+    /* Main loop */
+    for (count = 0; (sample_count == 0) || (count < sample_count); ++count) {
+        printf("Writing async, count %d\n", count);
+
+        /* Modify the data to be written here */
+        /* send count as data. */
+        instance->x = count;
+
+        /* Write data */
+        retcode =
+                asyncDataWriter_write(async_writer, instance, &instance_handle);
+        if (retcode != DDS_RETCODE_OK) {
+            printf("write error %d\n", retcode);
+        }
+
+        NDDS_Utility_sleep(&send_period);
+
+        retcode = DDS_DataWriter_get_datawriter_protocol_status(
+                writer,
+                &protocol_status);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "get_datawriter_protocol_status error\n");
+            publisher_shutdown(participant);
+            return -1;
+        }
+        printf("Asynchronous Publisher Statistics:\n"
```

Comment:
These are not all of the fragmented data statistics which are provided. I'm not sure if it would be helpful or confusing to including others (such as pushed_fragment_bytes)...What do you think?

Reply:
I have added:
```
        printf("Fragmented Data Statistics:\n"
                "\t pushed_fragment_count %lld\n"
                "\t pushed_fragment_bytes %lld\n"
                "\t pulled_fragment_count %lld\n"
                "\t pulled_fragment_bytes %lld\n"
                "\t received_nack_fragment_count %lld\n"
                "\t received_nack_fragment_bytes %lld\n",
                protocol_status.pushed_fragment_count,
                protocol_status.pushed_fragment_bytes,
                protocol_status.pulled_fragment_count,
                protocol_status.pulled_fragment_bytes,
                protocol_status.received_nack_fragment_count,
                protocol_status.received_nack_fragment_bytes);
```